src/thunderbots/software/ai/world/team.h

```diff
@@ -20,22 +21,62 @@ class Team
    public:
     /**
      * Create a new team
+     *
+     * @param robot_expiry_buffer_milliseconds The number of milliseconds a robot must not
+     * have been updated for before it is removed from the team
      */
-    explicit Team();
+    explicit Team(const std::chrono::milliseconds robot_expiry_buffer_milliseconds);
 
     /**
      * Updates this team with new robots.
      *
      * @param team_robots the new robots for this team
+     * @param timestamp The timestamp at which this update is taking place. The timestamp
+     * must be >= the last update timestamp of the robots on the team
+     */
+    void updateRobots(const std::vector<Robot>& team_robots,
+                      const std::chrono::steady_clock::time_point timestamp);
+
+    /**
+     * Updates this team with new data from the given team object
+     *
+     * @param new_team_data A team with the new team data
+     * @param timestamp The timestamp at which this update is taking place. The timestamp
+     * must be >= the last update timestamp of the robots on the team
+     */
+    void updateState(const Team& new_team_data,
+                     const std::chrono::steady_clock::time_point timestamp);
```

Comment:
How does this differ from a copy-constructor? If it doesn't, we should remove it, if it does, we should make that clear in the docstring.

Reply:
Different because it will also remove any expired robots. Will update docstring

Reply:
_Or_ we could leave it up to the caller to determine robot removal behavior, but I think combining it with the `update` functions is slightly simpler. Thoughts?

Reply:
decided to also remove the robot removal from these update functions. Updated docstring accordingly